### PR TITLE
Adds some money types.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

```SUMMARY: Content "Adds Free Merchant currency'"```

#### Purpose of change

This adds free merchant currency as discussed in https://github.com/CleverRaven/Cataclysm-DDA/issues/27955, https://github.com/CleverRaven/Cataclysm-DDA/issues/27449 and https://github.com/CleverRaven/Cataclysm-DDA/issues/27447

I do not believe it fully satisfies the latter two issues yet as this is just free merchants, with a small infrastructural change for the Campus mainly because I hope to soon add a few references to their existence amongst the merchants.

#### Describe the solution

Adds the Free Merchant Certified Note, commonly called the Merch. This is a promissory note printed on old US large-denom currency.

Currently the only way to get this is by bartering with the free merchant shopkeeper. I cannot see any way at this time to make the money less valuable outside the free merchant faction; the code will probably need to be updated to give a factional flag to currency allowing us to give it no or partial value to other factions.

I may add further support for the currency within this PR, eg selling food, but I need it added so that my NPCs can start asking for it.

#### Describe alternatives you've considered

There's been a long and involved discussion on various options for currency, but ultimately a promissory note is the most logical and this one has a good post apocalyptic flavour.

#### Additional context

Not yet checked in-game.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
You've got some unrelated changes in with this PR.

**Comment by I-am-Erk** [comment]
Yes, my local repository seems to be borked. I'm trying to fix it, but it's not updating... will probably have to restart this pr after I've figured it out

**Comment by Leland** [comment]
Suppose the PR is as good a place for this as any: the (standard game mode) CDDA starts 5 days after the Cataclysm. Is this currency one that the Free Merchants developed pre-Cataclysm? And if so, why?

I ask because developing a currency is no small thing (and there's likely been other things on peoples minds in the week after the apocalypse).

**Comment by I-am-Erk** [comment]
The Free Merchants would probably have worked out some form of promissory note within a few days of the Cataclysm; one of the reasons I like the signed bills is that they feel like something that would have started naturally and been rapidly adopted among them.

I assume that the lag between "end of the world" and "free merchants adopt promissory notes for internal allocation of resources" is short enough that there's not a lot of value in the number of conditionals it would take to simulate the transition period.

**Comment by AMurkin** [comment]
Maybe limit for some random period of time per world to rumors about new currency only? End game economics.

**Comment by I-am-Erk** [comment]
There's not really a ton of value to limiting it to a period of time. It adds a huge amount of work to every dialogue file related to paying for services, for the niche times a player makes it to the refugee center incredibly early or starts out there. Plus I currently can't limit item spawning in merchant inventories based on game time, so it would prevent me from letting the merchant trade for it.

File: data/json/items/generic/currency.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[
+  {
+    "type": "GENERIC",
+    "id": "FMCNote",
+    "name": "FMCN",
```
[the remark this answers is not in the document]
Comment (I-am-Erk) [reply data/json/items/generic/currency.json]: Fair call.

**Pull request merged**